### ISSUE TITLE
Handle workspce-relative paths in error messages

### ISSUE THREAD

**Issue by dherman** [dtolnay/trybuild#28]
Some error messages refer to absolute paths of filenames in libraries within a larger workspace, which can live outside off `$DIR`. For example, in [this failed test](https://travis-ci.org/neon-bindings/neon/jobs/590590804), the `*.stderr` error message generated by trybuild locally on my machine included:

```
error[E0277]: the trait bound `f64: std::convert::From<&str>` is not satisfied
   --> $DIR/example_compile_error.rs:6:5
    |
6   |     JsNumber::new(cx, "9000")
    |     ^^^^^^^^^^^^^ the trait `std::convert::From<&str>` is not implemented for `f64`
    | 
   ::: /Users/dherman/Sources/neon/src/types/mod.rs:313:39
    |
313 |     pub fn new<'a, C: Context<'a>, T: Into<f64>>(cx: &mut C, x: T) -> Handle<'a, JsNumber> {
    |                                       --------- required by this bound in `neon::types::JsNumber::new`
    |
```

This was part of a test that lives in a crate under `/Users/dherman/Sources/neon/tests/static`, which is part of a workspace rooted at `/Users/dherman/Sources/neon`. So that path lives outside of `$DIR` and doesn't get normalized. And the local paths don't match the paths on the CI machine.

Would it make sense to have a `$WORKSPACE` variable in addition to `$DIR` and `$CRATE`? Also, does the normalization also need to look at these `:::` lines?

**Comment by dherman** [comment]
@dtolnay I would be happy to try to implement a fix for this but would love your direction (and maybe tips on where to look in the code) before I go too far in any particular direction.

**Comment by dtolnay** [comment]
I released 1.0.14 with `$WORKSPACE` as you suggested. You should see:

```console
error[E0277]: the trait bound `f64: std::convert::From<&str>` is not satisfied
   --> $DIR/example_compile_error.rs:6:5
    |
6   |     JsNumber::new(cx, "9000")
    |     ^^^^^^^^^^^^^ the trait `std::convert::From<&str>` is not implemented for `f64`
    | 
   ::: $WORKSPACE/src/types/mod.rs:313:39
    |
313 |     pub fn new<'a, C: Context<'a>, T: Into<f64>>(cx: &mut C, x: T) -> Handle<'a, JsNumber> {
    |                                       --------- required by this bound in `neon::types::JsNumber::new`
    |
```
